nginx-controller/nginx/configurator.go

```
@@ -1013,9 +1013,41 @@ func mergeMasterAnnotationsIntoMinion(minionAnnotations map[string]string, maste
 	}
 }
 
+// GenerateNginxMainConfig generate NginxMainConfig from Config
+func GenerateNginxMainConfig(config *Config) *NginxMainConfig {
+	nginxCfg := &NginxMainConfig{
+		MainSnippets:              config.MainMainSnippets,
+		HTTPSnippets:              config.MainHTTPSnippets,
+		ServerNamesHashBucketSize: config.MainServerNamesHashBucketSize,
+		ServerNamesHashMaxSize:    config.MainServerNamesHashMaxSize,
+		LogFormat:                 config.MainLogFormat,
+		SSLProtocols:              config.MainServerSSLProtocols,
+		SSLCiphers:                config.MainServerSSLCiphers,
+		SSLDHParam:                config.MainServerSSLDHParam,
+		SSLPreferServerCiphers:    config.MainServerSSLPreferServerCiphers,
+		HTTP2:                 config.HTTP2,
+		ServerTokens:          config.ServerTokens,
+		ProxyProtocol:         config.ProxyProtocol,
+		WorkerProcesses:       config.MainWorkerProcesses,
+		WorkerCPUAffinity:     config.MainWorkerCPUAffinity,
+		WorkerShutdownTimeout: config.MainWorkerShutdownTimeout,
+		WorkerConnections:     config.MainWorkerConnections,
+		WorkerRlimitNofile:    config.MainWorkerRlimitNofile,
+	}
+	return nginxCfg
+}
+
 // UpdateConfig updates NGINX Configuration parameters
 func (cnf *Configurator) UpdateConfig(config *Config, ingExes []*IngressEx, mergeableIngs map[string]*MergeableIngresses) error {
 	cnf.config = config
+	if cnf.config.MainServerSSLDHParamFileContent != nil {
+		fileName, err := cnf.nginx.AddOrUpdateDHParam(*cnf.config.MainServerSSLDHParamFileContent)
+		if err != nil {
+			glog.Errorf("Could not update dhparams: %v", err)
```

let's return an error here